cipher-asymmetric/src/main/scala/tsec/cipher/asymmetric/imports/JCAAsymmetricCipher.scala

```diff
@@ -0,0 +1,68 @@
+package tsec.cipher.asymmetric
+
+import javax.crypto.{Cipher => JCipher}
+
+import tsec.cipher.asymmetric.core.AsymmetricCipherAlgebra
+import tsec.cipher.asymmetric.imports._
+import tsec.cipher.common
+import tsec.cipher.common._
+import tsec.cipher.common.mode.{ModeKeySpec, NoMode}
+import tsec.cipher.common.padding.Padding
+import cats.syntax.either._
+import tsec.common.ErrorConstruct._
+
+class JCAAsymmetricCipher[A, P](implicit algoTag: AsymmetricAlgorithm[A], paddingTag: Padding[P])
+    extends AsymmetricCipherAlgebra[Either[CipherError, ?], A, P] {
+
+  type C = JCipher
+
+  def genInstance =
+    Either
+      .catchNonFatal(JCipher.getInstance(s"${algoTag.algorithm}/None/${paddingTag.algorithm}", "BC"))
```

Comment:
This worries me.

ECIES is a little different than AES and whatnot. i.e [this](https://github.com/bcgit/bc-java/blob/master/prov/src/test/java/org/bouncycastle/jce/provider/test/ECIESTest.java#L139) or: [this](https://github.com/bcgit/bc-java/blob/master/prov/src/test/java/org/bouncycastle/jce/provider/test/ECIESTest.java#L184)

Do you think you could verify that Cipher.getInstance("ECIES", "BC") returns the same as what is in your interpreter? [This](https://github.com/google/wycheproof/blob/master/java/com/google/security/wycheproof/testcases/EciesTest.java#L122) states what you may be inputting could be an invalid name.